### PR TITLE
Added basic styling to Direct Debit and improved error handling

### PR DESCRIPTION
**Pull request by svillafe**
## Why are you doing this?

To improve the UX and UI of the direct debit form. **This is not the final design of the form**. There should be a design/UX iteration before the final release. 
[**Trello Card**](https://trello.com/c/nOdfGQgS/254-direct-debit-as-a-payment-mechanism)

## Changes

* Modify a11y label-for linting rule. Before it required both nesting and id link, now it only requires one of those.
* Come up with more meaningful codes for errors
* Added CSS✨ to improve the PopUp Form and the Direct debit form.

## Screenshots
directdebitpolished

#### Before
picture 521

#### After
picture 522

#### Before
picture 523

#### After
picture 524

### CODE REVIEW

**Comment by Amohkhan** [comment]
Almost there, agreed with Elvis! Thanks Santiago

File: assets/components/directDebit/directDebitForm/directDebitForm.scss [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  display: block;
+  font-family: $gu-sans-web;
+  font-size: 16px;
+  line-height: 24px;
+}
+
+.component-direct-debit-form__sort-code-field {
+  font-family: $gu-sans-web;
+  font-size: 14px;
+  color: gu-colour(neutral-1);
```
Comment (Ap0c) [inline assets/components/directDebit/directDebitForm/directDebitForm.scss]: @Amohkhan Should we be using `garnett-neutral-1` for new stuff?

File: assets/components/directDebit/directDebitForm/directDebitForm.scss [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  border-radius: 600px;
+  height: $gu-v-spacing * 4;
+  padding: 0 $gu-h-spacing;
+  width: 70px;
+}
+
+.component-direct-debit-form__text-field {
+  font-family: $gu-sans-web;
+  font-size: 14px;
+  color: gu-colour(neutral-1);
```
Comment (Ap0c) [inline assets/components/directDebit/directDebitForm/directDebitForm.scss]: @Amohkhan Same question as above.

File: assets/components/directDebit/directDebitForm/directDebitForm.scss [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  text-align: left;
+  width: 100%;
+  margin-top: $gu-v-spacing;
+
+  @include mq($from: mobileMedium) {
+    width: 175px;
+  }
+
+  &:hover {
+    background-color: #000;
```
Comment (Ap0c) [inline assets/components/directDebit/directDebitForm/directDebitForm.scss]: `darken(gu-colour(neutral-1), 10%)`

Or is it 5% now @Amohkhan ?

File: assets/components/directDebit/directDebitPopUpForm/directDebitPopUpForm.scss [hunk 4]
```diff
@@ -0,0 +1,44 @@
+.component-direct-debit-pop-up-form {
+  z-index: 7500;
```
Comment (Ap0c) [inline assets/components/directDebit/directDebitPopUpForm/directDebitPopUpForm.scss]: Why so large?

Comment (svillafe) [reply assets/components/directDebit/directDebitPopUpForm/directDebitPopUpForm.scss]: I put this number to match the scale used in `payment progress` message. We can review this practice and do a refactor in the future.

**Pull request merged**